packages/mattermost-redux/src/utils/post_list.ts

```diff
@@ -323,142 +322,100 @@ export function makeGenerateCombinedPost(): (state: GlobalState, combinedId: str
     );
 }
 
-export const postTypePriority = {
-    [Posts.POST_TYPES.JOIN_TEAM]: 0,
-    [Posts.POST_TYPES.ADD_TO_TEAM]: 1,
-    [Posts.POST_TYPES.LEAVE_TEAM]: 2,
-    [Posts.POST_TYPES.REMOVE_FROM_TEAM]: 3,
-    [Posts.POST_TYPES.JOIN_CHANNEL]: 4,
-    [Posts.POST_TYPES.ADD_TO_CHANNEL]: 5,
-    [Posts.POST_TYPES.LEAVE_CHANNEL]: 6,
-    [Posts.POST_TYPES.REMOVE_FROM_CHANNEL]: 7,
-    [Posts.POST_TYPES.PURPOSE_CHANGE]: 8,
-    [Posts.POST_TYPES.HEADER_CHANGE]: 9,
-    [Posts.POST_TYPES.JOIN_LEAVE]: 10,
-    [Posts.POST_TYPES.DISPLAYNAME_CHANGE]: 11,
-    [Posts.POST_TYPES.CONVERT_CHANNEL]: 12,
-    [Posts.POST_TYPES.CHANNEL_DELETED]: 13,
-    [Posts.POST_TYPES.CHANNEL_UNARCHIVED]: 14,
-    [Posts.POST_TYPES.ADD_REMOVE]: 15,
-    [Posts.POST_TYPES.EPHEMERAL]: 16,
-};
-
-export function comparePostTypes(a: typeof postTypePriority, b: typeof postTypePriority) {
-    return postTypePriority[a.postType] - postTypePriority[b.postType];
-}
-
-function extractUserActivityData(userActivities: any) {
+function extractUserActivityData(userActivities: ActivityEntry[]) {
     const messageData: any[] = [];
     const allUserIds: string[] = [];
     const allUsernames: string[] = [];
-    Object.entries(userActivities).forEach(([postType, values]: [string, any]) => {
-        if (
-            postType === Posts.POST_TYPES.ADD_TO_TEAM ||
-            postType === Posts.POST_TYPES.ADD_TO_CHANNEL ||
-            postType === Posts.POST_TYPES.REMOVE_FROM_CHANNEL
-        ) {
-            Object.keys(values).map((key) => [key, values[key]]).forEach(([actorId, users]) => {
-                if (Array.isArray(users)) {
-                    throw new Error('Invalid Post activity data');
-                }
-                const {ids, usernames} = users;
-                messageData.push({postType, userIds: [...usernames, ...ids], actorId});
-                if (ids.length > 0) {
-                    allUserIds.push(...ids);
+    userActivities.forEach((activity) => {
+        if (isUsersRelatedPost(activity.postType)) {
+            const {postType, actorId, userIds, usernames} = activity;
+            if (usernames && userIds) {
+                messageData.push({postType, userIds: [...usernames, ...userIds], actorId});
+                if (userIds.length > 0) {
+                    allUserIds.push(...userIds);
                 }
-
                 if (usernames.length > 0) {
                     allUsernames.push(...usernames);
                 }
                 allUserIds.push(actorId);
-            });
+            } else {
+                throw new Error('Invalid Post activity data');
+            }
         } else {
-            if (!Array.isArray(values)) {
+            const {postType, actorId} = activity;
+            const userIds = actorId;
```

Comment:
This probably isn't needed if we make the change I mentioned above, but this should be
```js
const userIds = [actorId];
```
to avoid having the `userIds` field of a `messageData` entry sometimes be `string` and other times be `string[]`